lib/Form/EventListener/SelectionMultilingualOptionsDataListener.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/**
+ * @copyright Copyright (C) eZ Systems AS. All rights reserved.
+ * @license For full copyright and license information view LICENSE file distributed with this source code.
+ */
+declare(strict_types=1);
+
+namespace EzSystems\RepositoryForms\Form\EventListener;
+
+use Symfony\Component\Form\FormEvent;
+
+class SelectionMultilingualOptionsDataListener
+{
+    /** @var string */
+    protected $languageCode;
+
+    /**
+     * @param string $languageCode
+     */
+    public function __construct(string $languageCode)
+    {
+        $this->languageCode = $languageCode;
+    }
+
+    /**
+     * @param \Symfony\Component\Form\FormEvent $event
+     */
+    public function setLanguageOptions(FormEvent $event): void
+    {
+        $languageOptions = [];
+        $data = $event->getData();
+        if (isset($data[$this->languageCode])) {
+            $languageOptions = $data[$this->languageCode];
+        }
+
+        $event->setData($languageOptions);
```

Comment:
Nitpick: how about
```php
$event->setData($data[$this->languageCode] ?? []);
```
instead of the above `if`?
But correct me if this changes behavior.